Tests/tests/test_RDKShell.cpp

```diff
@@ -0,0 +1,1041 @@
+                            "\"bounds\":{"
+                                "\"x\":0,"
+                                "\"y\":0,"
+                                "\"w\":1920,"
+                                "\"h\":1080"
+                            "},"
+                           "\"success\":true"
+                        "}"));
+}
+
+
+TEST_F(RDKShellTest, CursorSize)
+{
+    EXPECT_EQ(Core::ERROR_GENERAL, handler.Invoke(connection, _T("setCursorSize"), _T("{}"), response));
+    ON_CALL(compositormock, setCursorSize(::testing::_, ::testing::_))
+	    .WillByDefault(::testing::Invoke(
+                [](uint32_t width, uint32_t height){
+                      EXPECT_EQ(width, 255);
+                      EXPECT_EQ(height, 255);
+                      return true;
+                }));
+    EXPECT_EQ(Core::ERROR_NONE, handler.Invoke(connection, _T("setCursorSize"), _T("{\"width\":255,\"height\":255}"),response));
+    EXPECT_EQ(response, _T("{\"success\":true}"));
+
+    ON_CALL(compositormock, getCursorSize(::testing::_, ::testing::_))
+            .WillByDefault(::testing::Invoke(
+                [](uint32_t& width, uint32_t& height){
+                      width = 255;
+                      height = 255;
+                      return true;
+                }));
+
+    EXPECT_EQ(Core::ERROR_NONE, handler.Invoke(connection, _T("getCursorSize"), _T("{}"), response));
+    EXPECT_EQ(response, _T("{\"width\":255,\"height\":255,\"success\":true}"));
+}
+
+TEST_F(RDKShellTest, showandhideCursor)
+{
+    ON_CALL(compositormock, showCursor())
+         .WillByDefault(::testing::Return(true));
+    EXPECT_EQ(Core::ERROR_NONE, handler.Invoke(connection, _T("showCursor"), _T("{}"),response));
+    EXPECT_EQ(response, _T("{\"success\":true}"));
+    ON_CALL(compositormock, hideCursor())
+         .WillByDefault(::testing::Return(true));
+    EXPECT_EQ(Core::ERROR_NONE, handler.Invoke(connection, _T("hideCursor"), _T("{}"), response));
+    EXPECT_EQ(response, _T("{\"success\":true}"));
+}
+
+TEST_F(RDKShellTest, splashLogo)
+{
+    EXPECT_EQ(Core::ERROR_GENERAL, handler.Invoke(connection, _T("showSplashLogo"), _T("{}"),response));
+    EXPECT_EQ(Core::ERROR_NONE, handler.Invoke(connection, _T("showSplashLogo"), _T("{\"displayTime\": 5}"),response));
+    EXPECT_EQ(response, _T("{\"success\":true}"));
+    ON_CALL(compositormock, hideSplashScreen())
+         .WillByDefault(::testing::Return(true));
+    ON_CALL(RDKShellmock, logMilestone(::testing::_))
+            .WillByDefault(::testing::Invoke(
+                [](const char *msg_code){
+                }));
+
+    EXPECT_EQ(Core::ERROR_NONE, handler.Invoke(connection, _T("hideSplashLogo"), _T("{}"),response));
+    EXPECT_EQ(response, _T("{\"success\":true}"));
+    ON_CALL(compositormock, hideSplashScreen())
+         .WillByDefault(::testing::Return(false));
+    EXPECT_EQ(Core::ERROR_GENERAL, handler.Invoke(connection, _T("hideSplashLogo"), _T("{}"),response));
+}
```

Comment:
I don't think such test makes sense
how do you verify that "showSplashLogo" shows/hides the logo?
how do you verify displayTime has effect?

Reply:
it will set gSplashScreenDisplayTime = displayTime and  receivedShowSplashScreenRequest = true
and while plugin initializing    it will check  receivedShowSplashScreenRequest and calls CompositorController::showSplashScreen in RdkShell component